src/components/Dialogs/FixedDimension.js

```diff
@@ -70,8 +88,97 @@ const FixedDimension = ({
 
     const closeModal = () => onClose()
 
+    const renderStatusParagraph = () => (
+        <p className={classes.paragraph}>
+            {i18n.t('Show items where the status is:')}
+        </p>
+    )
+
+    const setStatus = ({ dimensionId, selectedItemsIds, itemId, toggle }) => {
+        const newIds = toggle
+            ? [...new Set([...selectedItemsIds, itemId])]
+            : selectedItemsIds.filter((id) => id !== itemId)
+
+        selectUiItems({
+            dimensionId,
+            items: newIds.map((id) => ({ id })),
+        })
+    }
+
+    const renderProgramStatus = () => {
+        const ALL_STATUSES = [
+            { id: STATUS_ACTIVE, name: statusNames[STATUS_ACTIVE] },
+            { id: STATUS_COMPLETED, name: statusNames[STATUS_COMPLETED] },
+            { id: STATUS_CANCELLED, name: statusNames[STATUS_CANCELLED] },
+        ]
+
+        return (
+            <>
+                {renderStatusParagraph()}
+                <div>
+                    {ALL_STATUSES.map(({ id, name }) => (
+                        <Checkbox
+                            key={id}
+                            checked={programStatusIds.includes(id)}
+                            label={name}
+                            onChange={({ checked }) =>
+                                setStatus({
+                                    dimensionId: DIMENSION_TYPE_PROGRAM_STATUS,
+                                    selectedItemsIds: programStatusIds,
+                                    itemId: id,
+                                    toggle: checked,
+                                })
+                            }
+                            dense
+                            className={classes.verticalCheckbox}
+                        />
+                    ))}
+                </div>
+            </>
+        )
+    }
+
+    const renderEventStatus = () => {
```

Comment:
Yeah I considered that as well, but I figured since we'll never have more than these two and since the amount of common lines are quite low compared to the unique stuff I thought it would just over-complicate things (even though there's clearly a common pattern here)